imap_data_access/io.py

```diff
@@ -61,16 +61,19 @@ def download(file_path: Union[Path, str]) -> Path:
         Path to the downloaded file
     """
     destination = imap_data_access.config["DATA_DIR"]
-    if isinstance(file_path, str) and "/" not in file_path:
-        # Construct the directory structure from the filename
-        # This is for science files that contain the directory structure in the filename
-        # Otherwise, we assume the full path to the file was given
+    # Create the proper file path object based on the extension and filename
+    file_path = Path(file_path)
+    if file_path.suffix[1:] in imap_data_access.VALID_SPICE_EXTENSIONS:
+        # SPICE
+        path_obj = imap_data_access.SPICEFilePath(file_path.name)
+    else:
+        # Science
+        path_obj = imap_data_access.ScienceFilePath(file_path.name)
 
-        science_file_path = imap_data_access.ScienceFilePath(file_path)
+    destination = path_obj.construct_path()
 
-        destination = science_file_path.construct_path()
-    else:
-        destination /= file_path
+    # Update the file_path with the full path for the download below
+    file_path = str(destination.relative_to(imap_data_access.config["DATA_DIR"]))
```

Comment:
This is the key to be able to download a file with either the filename, or the full path. Make sure we make it a full relative path before sending the request.